modules/api/pkg/handler/v2/cluster_default/cluster_default.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterdefault
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/gorilla/mux"
+
+	apiv1 "k8c.io/dashboard/v2/pkg/api/v1"
+	apiv2 "k8c.io/dashboard/v2/pkg/api/v2"
+	"k8c.io/dashboard/v2/pkg/handler/middleware"
+	"k8c.io/dashboard/v2/pkg/handler/v1/common"
+	clusterv2 "k8c.io/dashboard/v2/pkg/handler/v2/cluster"
+	"k8c.io/dashboard/v2/pkg/handler/v2/networkdefaults"
+	"k8c.io/dashboard/v2/pkg/provider"
+	kubermaticprovider "k8c.io/dashboard/v2/pkg/provider"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/defaulting"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	utilerrors "k8c.io/kubermatic/v2/pkg/util/errors"
+
+	"k8s.io/utils/pointer"
+)
+
+// getDefaultClusterReq represents a request for retrieving a default spec for the cluster.
+// swagger:parameters getDefaultCluster
+type getDefaultClusterReq struct {
+	// in: path
+	// required: true
+	ProviderName string `json:"provider_name"`
+	// in: path
+	// required: true
+	DC string `json:"dc"`
+
+	// private field for the seed name. Needed for the cluster provider.
+	seedName string
+}
+
+// GetSeedCluster returns the SeedCluster object.
+func (req getDefaultClusterReq) GetSeedCluster() apiv1.SeedCluster {
+	return apiv1.SeedCluster{
+		SeedName: req.seedName,
+	}
+}
+
+// Validate validates getDefaultClusterReq request.
+func (r getDefaultClusterReq) Validate() error {
+	if r.ProviderName == "" {
+		return fmt.Errorf("the provider name cannot be empty")
```

Comment:
I'm not sure if returning an error here results in a proper http code. As those parameters are parts of path, it probably  doesn't matter, but I think explicitly returning bad request in validation is a good pattern.
```suggestion
		return utilerrors.NewBadRequest("the provider name cannot be empty")
```

Reply:
It will be transformed to `400` behind the curtains. That's how validation for requests works ;)